yellowbrick/text/postag.py

```diff
@@ -20,125 +20,371 @@
+                    elif tag == "CCONJ":
+                        self.tag_map["conjunction"] += 1
+                    elif tag == "PUNCT":
+                        self.tag_map["punctuation"] += 1
+                    elif tag == "DET":
+                        self.tag_map["determiner"] += 1
+                    elif tag == "NUM":
+                        self.tag_map["number"] += 1
+                    elif tag == "INTJ":
+                        self.tag_map["interjection"] += 1
+                    elif tag == "SYM":
+                        self.tag_map["symbol"] += 1
+                    elif tag == "SPACE":
+                        continue
+                    else:
+                        self.tag_map["other"] += 1                
+                
+    def _handle_treebank(self, X):
+        """
+        Create a part-of-speech tag mapping using the Penn Treebank tags
+        
+        Parameters
+        ----------
+        X : list or generator
+            Should be provided as a list of documents or a generator
+            that yields a list of documents that contain a list of 
+            sentences that contain (token, tag) tuples.
+        """
+        for tagged_doc in X:
+            for tagged_sent in tagged_doc:
+                for _, tag in tagged_sent:
+                    if tag.startswith("N"):
+                        self.tag_map["noun"] += 1
+                    elif tag.startswith("J"):
+                        self.tag_map["adjective"] += 1
+                    elif tag.startswith("V"):
+                        self.tag_map["verb"] += 1
+                    # include particles with adverbs
+                    elif tag.startswith("RB") or tag == "RP":
+                        self.tag_map["adverb"] += 1
+                    elif tag.startswith("PR"):
+                        self.tag_map["pronoun"] += 1
+                    elif tag.startswith("W"):
+                        self.tag_map["wh- word"] += 1
+                    elif tag == "CC":
+                        self.tag_map["conjunction"] += 1
+                    elif tag == "CD":
+                        self.tag_map["digit"] += 1
+                    # combine predeterminer and determiner
+                    elif tag in ["DT" or "PDT"]:
+                        self.tag_map["determiner"] += 1
+                    elif tag == "EX":
+                        self.tag_map["existential"] += 1
+                    elif tag == "FW":
+                        self.tag_map["non-English"] += 1
+                    elif tag == "IN":
+                        self.tag_map["preposition"] += 1
+                    elif tag == "POS":
+                        self.tag_map["possessive"] += 1
+                    elif tag == "LS":
+                        self.tag_map["list"] += 1
+                    elif tag == "MD":
+                        self.tag_map["modal"] += 1
+                    elif tag in self.punct_list:
```

Comment:
See note above about punctuation list, which should be easy enough to adapt to the alternate formulation if required. If more extensive punctuation checking is required, e.g. for unicode, let me know and I can supply a `unicodedata.category` method of punctuation checking.

Reply:
See above response, but NLTK's implementation of Penn Treebank automatically converts all sentence closers to periods. For now, these tags only get used for Penn Treebank (SpaCy turns everything to `PUNCT`). I imagined that this may change as other tagsets got included? To ensure no one gets confused about this, I've added a note and changed the var name per your suggestion.